lib/web/usertasks.go

```diff
@@ -113,14 +113,29 @@ func (h *Handler) userTaskListByIntegration(w http.ResponseWriter, r *http.Reque
 		return nil, trace.BadParameter("integration query param is required")
 	}
 
-	userTasks, nextKey, err := clt.UserTasksServiceClient().ListUserTasksByIntegration(r.Context(), int64(limit), startKey, integrationName)
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-
-	items := make([]ui.UserTask, 0, len(userTasks))
-	for _, userTask := range userTasks {
-		items = append(items, ui.MakeUserTask(userTask))
+	taskStateFilter := values.Get("state")
+
+	var nextKey string
+	items := make([]ui.UserTask, 0, limit)
+	pageLimit := limit
+	for {
+		userTasks, nextKey, err := clt.UserTasksServiceClient().ListUserTasksByIntegration(r.Context(), int64(pageLimit), startKey, integrationName)
```

Comment:
This shadows your previous declaration of nextKey. Is that what you intended?

Reply:
That was a bug. Thank you.
I intended to do pagination tests but I wasn't actually testing more than 1 page.
Fixed the code, and also the test to ensure we hit at least two pages.